### PR TITLE
Make libvirtError public to allow clients to discern between different error types

### PR DESCRIPTION
**Pull request by dmacvicar**
This PR supersedes https://github.com/digitalocean/go-libvirt/pull/123 incorporating the feedback given in that PR.

This PR is needed to continue the experimental migration of terraform-provider-libvirt to pure-go, in order to be able to use the Terraform registry. https://github.com/dmacvicar/terraform-provider-libvirt/pull/813

What was done:

* Rename the generated `libvirt.Error` to `libvirt.remote_error`
* Rename libvirt.libvirtError to `libvirt.Error`
* Expose all Error enum values
* Expose only error Code and Message
  (I am not convinced this is a good idea, but it was given as a guideline)

- I regenerated the files against libvirt 7.1.0

Note:

- I could still make `.Message` private, as it is accessible via `.Error()` too

// cc @MalloZup @kskewes @benlemasurier  @trapgate @vtolstov

### CODE REVIEW

**Comment by MalloZup** [comment]
@trapgate or your team/maintainers any chance you could review this ? that would help us a lot to move forward.. thank you in advance

File: remote_protocol.gen.go
Comment (trapgate) [inline remote_protocol.gen.go:194]: I looked into this change. The generator isn't doing the right thing here - you wouldn't expect a parameter _named_ "Error" to be renamed "remote_error". This is because the code generator passes both identifier names and type names into the same routine to be transformed. There are other symptoms of this in the generated code today: for example, a number of struct fields get renamed from "Secret" to "OptSecret", which is a type name.

I do want to fix this problem, but as it predates your PR and will be an API-breaking change (because some public struct field names will change), I'll do it as a separate PR after yours has merged.

**Review by trapgate — COMMENTED** [review]
One change, but this looks good to me otherwise.

File: libvirt.yml
Comment (dmacvicar) [reply libvirt.yml:59]: I have addressed this.

**Review by trapgate — APPROVED** [review]
This looks good to me. Thank you again for your work getting this done!

I'll merge this shortly.

**Comment by MalloZup** [comment]
thx @dmacvicar @trapgate

**Pull request merged**
